File: Dynamic_Programming_Rod_Cutting/Dynamic_Programming_Rod_Cutting.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Dynamic_Programming_Rod_Cutting
{
    class Program
    {

        private static int max(int a , int b )
        {

            return (a > b) ? a : b;
        }
        private static int money( int []arr, int size)
        {
            if( size <= 0 )
            return 0 ;
            else
            {
                int max_val = Int32.MinValue;
                for (int i = 0; i < size; i++)
                    max_val = max(max_val, arr[i] + money(arr, size - i - 1));
                return max_val;
            }

            
        }
        static void Main(string[] args)
        {
            int [] arr = {1, 5, 8, 9, 10, 17, 17, 20};
            int size = arr.Length;
            Console.Write("MAximum Cost : " + money(arr, size));
            Console.Read();
            //output 22

        }
    }
}

File: Dynamic_Programming_Rod_Cutting/Dynamic_Programming_Rod_Cutting.py

INT_MIN = -32767

# Function for money calculation
def money(price, n):
    val = [0 for x in range(n + 1)]
    val[0] = 0

    for i in range(1, n + 1):
        max_val = INT_MIN
        for j in range(i):
             max_val = max(max_val, price[j] + val[i - j - 1])
        val[i] = max_val

    return val[n]

# Testing the function
size = int(input())
arr = []
for i in range(0, size):
    arr.append(int(input()))
print("Maximum Money : " + str(money(arr, size)))

'''
INPUT: 
size = 8
arr = {3, 5, 8, 9, 10, 17, 17, 20}
OUTPUT:
Maximum value is 24
'''
